### DISCUSSION TITLE
How to find the seed from an image generated in a batch?

### DISCUSSION THREAD

**Discussion by yushan777** [comfyanonymous/ComfyUI#1124] [Q&A]
When images are generated in a batch, they all contain the same prompt/workflow data to be able to reproduce that batch and all contain the initial seed.  What it does not contain is the individual seed unique to that image. So let's say out of a batch of 100 images, I like image 77 and wish to reproduce that one and experiment.  there's no way for me to do that since when I drag/drop the image into comfyUI, it just sets up the workflow to generate that original batch again.   I initially thought that all images in a batch are just a +1 increment from the initial seed, but this does not appear to be so.  How do you find the individual seed for an image that was produced in a batch so you can say, reproduce that single image in isolation?

**Comment by ntdviet** [comment]
I had the same question. And until I get the answer, my work around is using the ExtraOptions / BatchCount instead

**Reply by yushan777** [reply]
Thank you for your response. Can you elaborate on this?  Is ExtraOptions a set of custom nodes or something else?

**Reply by ntdviet** [reply]
Here 
![ảnh](https://github.com/comfyanonymous/ComfyUI/assets/54492570/895135e1-1b9d-4cf9-a364-60d1077c6bd6)
If you set batch_size = N at the EmptyLatentImage, each and every Sampler node in your workflow will process N latents at once, before all the latent images move further down the line. When your workflow is finished, all N images are saved to your drive. This requires a lot of memory.

Whileas if you set batch_size = 1 at the EmptyLatentImage node, and set ExtraOptions/BatchCount = N, your workflow will be repeated N times and for every round, it spits out an image, 1 by 1. This require less memory and every picture generated get it's own Seed saved.

**Comment by BlenderNeko** [comment]
All required randomness in a batch of images is derived from that seed. If you want to isolate a single image in that batch use the [Latent From Batch](https://blenderneko.github.io/ComfyUI-docs/Core%20Nodes/Latent/batch/LatentFromBatch/) node. You can put this node **before** the sampler to make the sampler generate just that selected image.

**Reply by yushan777** [reply]
Thank you for your response.  Just so I understand, would the batch need to be run again with this `Latent From Batch` node used to intercept a specific generation?

**Reply by BlenderNeko** [reply]
The trick is to use that node before anything expensive is going to happen to batch. If you e.g. put it before any of the samplers, the sampler will only keep itself busy with generating the images you picked with `Latent From Batch`. So your entire workflow and all of the settings will look the same (including the batch count), the only difference is that you have a `Latent From Batch` inserted somewhere.

**Reply by Wislos** [reply]
I came here to report a bug and saw this comment. I think "Latent from Batch" is bugged. 
I wanted to make a batch and then cherry pick one image for a new round of txt2img. Paranoid as I am (and because I didnt know if this is even the right node for the job) I first tried to replicate the image. It doesn't work when you set the "length" to 1! It is slightly different (sometimes just a few details/textures, sometimes whole poses).
Weirdly enough it works when you set the "lenght" to higher than one but then I have to render two images if I just want one.
Does somebody know a solution or should I report it as a bug?

**Reply by ntdviet** [reply]
Thanks for the hint @BlenderNeko

**Reply by BlenderNeko** [reply]
@Wislos It seems that ancestral and SDE samplers don't play nice with latent from batch, You can open an issue about this if you want.

**Reply by Wislos** [reply]
Oh yeah just tested it, it is even worse for ancestral and SDE, even having the lenght > 1 doesn't work

**Reply by belladoreai** [reply]
@BlenderNeko  I don't understand how Latent From Batch can help you recover the seed of an individual image from within a batch? I get the latents corresponding to that individual image from a batch, but I also need to pass the correct seed to the KSampler, and it doesn't seem like that's possible with Latent From Batch that you suggested?

**Comment by wowop2000** [comment]
you can use the "xy plot " node,and connect the "xy input:seed++batch" node,all these nodes include by [Efficiency Nodes](https://github.com/LucianoCirino/efficiency-nodes-comfyui).

**Reply by BensonKWT** [reply]
perfect!

**Comment by Dreamer4** [comment]
@Wislos You should put the same seed from your generated images to your new sampler and set denoise to 0.01. Note that batch count starts from 1 but "Latent From Batch" starts from 0

**Comment by ewebgh33** [comment]
I came here via search looking for the same answer.
In Auto1111 I can XY Plot as many images as can fit into a PNG or JPG. 30? 50? I don't know maybe 100. But each one of those, contains the metadata for the settings.
In Comfy I was testing CFG scale effect on a few alternate workflows including for Flux, and when I loaded the images I liked back into ComfyUI... it doesn't show the CFG that generated THAT image. It loads the workflow that includes the batch starting from Zero.

Is "metadata" for the ACTUAL image, included in the image? If not, why not, why was that decision made?
Do we have to use a custom node to save an image with all the different metadata we would "normally" expect from an image generation?

Here I distinguish between the workflow/setup and the shorter metadata.
I understand workflows in Comfy can be more complex. We could have 2, 3, 4 or more Samplers and multiple processes.
But say, at least the LAST Sampler settings into metadata as a default?

If I can't tell what params made the images I was pleased with, how can I replicate? I feel like I've wasted several hours.
For now my workaround is to bake the info into the images themselves using a label node, but this is not ideal. I would rather be able to query metadata in the image file itself.

Thanks

**Reply by belladoreai** [reply]
I wasted a lot of time on this and my recommendation to you is to not use image batches.

**Reply by caniyabanci76** [reply]
Same as @belladoreai , I only use the queue batch option under Extra Options from the Menu panel.  This ensures each generation has its own unique seed and params.

**Reply by ewebgh33** [reply]
Well, I am using the the queue batch option 
So I set it to generate 10 times, 1 after the other (yeah, not latent batch)

To increment numbers of my choice, I have a seed node set to increment after each generation. I am using the seed node as an integer or float, that I pipe into something like CFG. Test 1-10 etc. I also convert to a string and pass into an Add Label node.

But each image still loads the workflow with the seed node set to increment from zero, not what it would have been set to for THAT image, if that makes sense

**Comment by ltdrdata** [comment]
![latent_from_batch](https://github.com/user-attachments/assets/b879c9f9-4d0f-486d-8a38-1a72f6195e2a)

When using batch latents in ComfyUI, noise is not applied separately to each latent image, but rather the entire latent tensor is initialized at once.
There is no such thing as a seed for the nth batch.
Therefore, you must use `Latent From Batch`.

**Reply by ewebgh33** [reply]
Thanks!
Personally - not what I was asking (speaking only for myself here).
Generally - An important point! Thanks for adding it. Batch latents is convenient, but not the best way to go for non-casual users.

To me the main flaw seems to be that while the workflow is saved in each image, it doesn't save seeds that generated each image specifically, within the workflow saved in the PNG for THAT image.

I can't think of a great way to do that though, unless it's via a checkbox in settings, say "force seed replacement in batch", which would replace the starting seed(s) in any workflow that happens to use increments or other such methods to derive a seed or other dynamic value.

So I suppose it should be "force dynamic value replacement", not seed replacement.

EG: The tickbox explains the caveat that the workflow saved inside the PNG, will then be the workflow _for that image only_ and not the _general_ workflow that was used to kick off a sequential set of renders or otherwise mathematically/incremented values.

Sounds complicated though. So for now I rely on annotation and burning in labels to images.

**Reply by belladoreai** [reply]
Thanks @ltdrdata ! Looks like this resolves the use case I was thinking of, which is to reproduce a single image from a batch.

**Comment by wibur0620** [comment]
I noticed that no one has explained the reason for this issue, so let me clarify. The reason for this is that the diffusion model is a probabilistic generation model. It selects the images that are most probabilistically close to the user's input requirements for display, despite having the same seed. Even if you need ten images at once, it will decode the ten images that are closest to your requirements after a single diffusion generation process.
